### ISSUE TITLE
Improve search functions for Hosts, Services and Notifications

### ISSUE THREAD

**Issue by mwaldmueller** [Icinga/icingaweb2-module-director#1672]
## Expected Behavior
* Consideration of contents from custom properties at Hosts,Services and Notifications
* Filter for custom properties and contents
* Global search fields at Services, Commands and Notifcations

## Current Behavior
* Names (Hosts)
* Names and assign rules (Services and Notifications)

## Your Environment

* Director version (System - About): v1.5.1
* Icinga Web 2 version and modules (System - About): 2.6.1
* Icinga 2 version (`icinga2 --version`): 2.9.2
* Operating System and version: CentOS 7.5
* Webserver, PHP versions: Apache 2.4.6, PHP 7.1.8

**Comment by Thomas-Gelf** [comment]
When i correctly understood the requirement, this depends on #573. That feature has initially been postponed as we developed similar functionality for IcingaDB and wanted to avoid redundant code. As IcingaDB is currently developed in Go, this concern is no longer true and we could address this topic.

We already implemented important building blocks for this, still, it's a fat feature. I'd really love to see it implemented, as it would make many things easier, faster - and offer a broad range of new possibilities. This should however be well planned and implemented in a dedicated block.
